src/Sensor.cc

```diff
@@ -551,3 +608,81 @@ bool Sensor::HasConnections() const
 {
   return true;
 }
+
+//////////////////////////////////////////////////
+bool Sensor::EnableTriggered(const std::string &_trigger_topic) {
+  return this->dataPtr->EnableTriggered(_trigger_topic);
+}
+
+//////////////////////////////////////////////////
+bool SensorPrivate::EnableTriggered(const std::string &_topic) {
+  if (_topic.empty())
+  {
+    gzwarn << "Trigger topic is empty for sensor [" << this->name << "]"
+           << std::endl;
+    return false;
+  }
+  if (!this->node.Subscribe(_topic, &SensorPrivate::OnTrigger, this))
+  {
+    gzwarn << "Failed to subscribe to trigger topic [" << _topic
+           << "] for sensor [" << this->name << "]" << std::endl;
+    return false;
+  }
+  this->triggerTopic = _topic;
+
+  // Set update rate to 0 so that sensor will update immediately on pending
+  // trigger.
+  std::lock_guard<std::mutex> lock(this->updateRateMutex);
+  this->updateRate = 0.0;
```

Comment:
do we still need to set updateRate to 0? The sensor will be forced to update (with `force` set to `true`) if there is a `pendingTrigger`?

Reply:
We don't need to set updateRate to 0, but a modification is needed in gz::sim::Sensors system to call `manager.RunOnce()` immediately if a sensor is triggered. I'll open a PR to address this.